lib/src/units_of_measurement/shared.dart

```diff
@@ -784,3 +783,48 @@ extension _StringExtensions on String {
     return result;
   }
 }
+
+// I hate to expose this even with @visibleForTesting, but I really want to test its logic directly rather than
+// requiring mental gymnastics to determine what inputs to feed into formatting a rated unit of measurement to exercise
+// the logic within this class.
+@visibleForTesting
+class ComparableRatedValue implements Comparable<ComparableRatedValue> {
```

Comment:
In implementing this work, I noticed some pretty awful behavior when formatting rates. Basically, the heuristic for determining which rate unit to use was poor. I decided to enhance the behavior drastically as part of this PR. The short of it is that this `ComparableRatedValue` class is used to compare rated values for "most human-readable", allowing a simple sort to choose the best option during formatting.